src/lib.ts

```diff
@@ -109,3 +112,62 @@ export function createResponseType(
 
   return `export type ${pascaleName}Response${genericArgsWithDefaults} = Required<${pascaleName}Record${genericArgsForRecord}> & ${systemFields}${expandArgString}`
 }
+
+function createFieldConstraints(
+  name: string,
+  schema: Array<FieldSchema>
+): string {
+  const fieldsToIgnore = ["id", "password", "tokenKey", "token"]
+  const textBasedTypes = ["text", "email", "url", "number"]
+
+  const constraints = schema
+    .filter((field) => {
+      if (field.system || fieldsToIgnore.includes(field.name)) return false
+      if (!textBasedTypes.includes(field.type)) return false
+      const hasMin = typeof field.min === "number" && field.min !== 0
+      const hasMax = typeof field.max === "number" && field.max !== 0
+      return hasMin || hasMax
+    })
+    .map((field) => {
+      const parts: string[] = []
+      if (typeof field.min === "number" && field.min !== 0) {
+        parts.push(`min: ${field.min}`)
+      }
+      if (typeof field.max === "number" && field.max !== 0) {
+        parts.push(`max: ${field.max}`)
+      }
+
+      if (parts.length === 0) return null
+      return `  ${field.name}: { ${parts.join(", ")} }`
+    })
+    .filter(Boolean)
+
+  if (constraints.length === 0) return ""
+
+  const typeName = toPascalCase(name)
+  return `export const ${typeName}FieldConstraints = {
+${constraints.join(",\n")}
+}`
+}
+
+export function generateConstraints(results: Array<CollectionRecord>): string {
```

Comment:
Lets match the existing pattern by keeping this function here, but put `createFieldConstraints` in a new file "metadata.ts"